data-reconciliation-app/src/endpoints/reporting.ts

```diff
@@ -0,0 +1,31 @@
+import * as ccfapp from "@microsoft/ccf-app";
+import { ServiceResult } from "../utils/service-result";
+import { reportingService, authenticationService } from "../utils/dependencies";
+import { ApiResult } from "../utils/api-result";
+
+export function getDataSummary(
+  request: ccfapp.Request<any>
+): ccfapp.Response<any> {
+  try {
```

Comment:
Please see my comment above. The try/catch is best for dependencies and expressions that can throw an exception. Any expression returning a ServiceResult doesn't require to be inside the try/catch block. ServiceResult pattern is built on the convention of service handling all the known exceptions in the control flow.